### DISCUSSION TITLE
Evaluating Core for Mobile Ad-hoc Networks

### DISCUSSION THREAD

**Discussion by chief17** [coreemu/core#880] [General]
Hello All,

I am currently evaluating Core emulator for my emulation purposes of a Mobile Wireless Ad-hoc Network.

I aim to Create a Wireless Mesh Network - I am hoping to run Batman-adv to implement this on nodes. ( Till now I don't see Core's capability to deploy L2 Network devices (access points/NAT/Bridges)??).

Is there a way to emulate this behaviour?

**Comment by shanv82** [comment]
Did you figure this out ? I have a similar requirement - to have CORE nodes connect/disconnect with different networks based on CORE access point nodes.

**Reply by chief17** [reply]
Due to lack of documentation, i switched to a different platform. But I am still hopeful that with time more explaination and documentation will be added to make this viable for diverse usecases. 😔

**Reply by cvs1111** [reply]
This might have helped: https://coreemu.github.io/core/tutorials/tutorial2.html

and you can run batman inside a virtual core node as a custom service :  https://coreemu.github.io/core/services.html

**Comment by bharnden** [comment]
CORE is just the glue to tie things together, more or less. Whatever software you can run and whatever it can do is possible, but up to the user to dictate. It would be impossible to try and cover every base, for every person, with all possible/potential configurations etc.

COREs Switches/Hubs are Linux Bridges and can be swapped with OVS bridges, if OVS is installed and `core-daemon` is ran with `--ovs`.

So anything you can run, for whatever reason that may be, can potentially work.

If there is something specific desired, you can always bring that up for consideration for the future. Most of the default `nodes` in CORE use `services` which just run arbitrary shell commands and provided as a level of convenience, but not meant to be end all answers.
